File: 0258-27-3066-minimum-operations-to-exceed-threshold-value-ii/0258-27-3066-minimum-operations-to-exceed-threshold-value-ii.py
import heapq

class Solution:
    def minOperations(self, nums, k):
        heapq.heapify(nums)
        cnt = 0

        while len(nums) > 1 and nums[0] < k:
            x = heapq.heappop(nums)
            y = heapq.heappop(nums)
            heapq.heappush(nums, min(x * 2 + y, k + 1))
            cnt += 1

        return cnt

File: 0258-27-3066-minimum-operations-to-exceed-threshold-value-ii/0258-27-3066-minimum-operations-to-exceed-threshold-value-ii.java
import java.util.PriorityQueue;

class Solution {
    public int minOperations(int[] nums, int k) {
        PriorityQueue<Long> pq = new PriorityQueue<>();
        
        for (int num : nums) {
            pq.offer((long) num);
        }

        int cnt = 0;

        while (pq.size() > 1 && pq.peek() < k) {
            long x = pq.poll();
            long y = pq.poll();
            pq.offer(Math.min(x * 2 + y, (long) k + 1)); // Ensure long multiplication
            cnt++;
        }

        return cnt;
    }
}
